src/ttsim/plot_dag.py

```diff
@@ -1,150 +1,316 @@
+            type=node_selector.type,
+            order=node_selector.order,
         )
-    fig = _plot_dag(dag=dag, title=title)
-    if output_path.suffix == ".html":
-        fig.write_html(output_path)
     else:
-        raise ValueError(f"Unsupported file extension: {output_path.suffix}")
-
-    if with_params:
-        f = dags.concatenate_functions(
-            dag=dag,
-            functions=nodes,
-            targets=targets,
-            return_type="dict",
-            enforce_signature=False,
-            set_annotations=False,
-        )
-        args = dict(inspect.signature(f).parameters)
-        args.pop("backend", None)
-        args.pop("xnp", None)
-        args.pop("dnp", None)
-        if args:
-            raise ValueError(
-                "The policy environment DAG should include all root nodes but requires "
-                f"inputs:\n\n{format_list_linewise(args.keys())}",
-            )
+        qname_node_selector = None
+
+    dag_with_node_metadata = _get_tt_dag_with_node_metadata(
+        environment=environment,
+        node_selector=qname_node_selector,
+        include_params=include_params,
+    )
+    return _plot_dag(
+        dag=dag_with_node_metadata,
+        title=title,
+        show_node_description=show_node_description,
+    )
 
 
-def plot_full_interface_dag(output_path: Path) -> None:
+def plot_interface_dag(show_node_description: bool = False) -> go.Figure:
     """Plot the full interface DAG."""
     nodes = {
-        p: n.dummy_callable() if isinstance(n, InterfaceInput) else n
+        p: dummy_callable(n) if not callable(n) else n
         for p, n in load_interface_functions_and_inputs().items()
     }
-
     dag = dags.create_dag(functions=nodes, targets=None)
-    f = dags.concatenate_functions(
+
+    for name, node_object in nodes.items():
+        f = node_object.function if hasattr(node_object, "function") else node_object
+        description = inspect.getdoc(f) or "No description available."
+        namespace = name.split("__")[0] if "__" in name else "top-level"
+        dag.nodes[name]["node_metadata"] = NodeMetaData(
+            description=description,
+            namespace=namespace,
+        )
+
+    return _plot_dag(
         dag=dag,
-        functions=nodes,
-        targets=None,
-        return_type="dict",
-        enforce_signature=False,
-        set_annotations=False,
+        title="Full Interface DAG",
+        show_node_description=show_node_description,
     )
-    args = inspect.signature(f).parameters
-    if args:
-        raise ValueError(
-            "The full interface DAG should include all root nodes but requires inputs:"
-            f"\n\n{format_list_linewise(args.keys())}",
+
+
+def _get_tt_dag_with_node_metadata(
+    environment: NestedPolicyEnvironment,
+    node_selector: _QNameNodeSelector | None = None,
+    include_params: bool = True,
+    include_other_objects: bool = True,
+) -> nx.DiGraph:
```

Comment:
Please propagate to the main interface. This will exclude stuff like the backend, num_segments, etc.